### PR TITLE
Add slice data functional tests

### PR DESCRIPTION
**Pull request by joyceerhl**
For #5066

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py [hunk 1]
```diff
@@ -91,14 +91,14 @@ def _VSCODE_convertTensorToDataFrame(tensor, start=None, end=None):
...
     elif isinstance(df, _VSCODE_pd.Series):
-        df = _VSCODE_pd.Series.to_frame(df)
+        df = _VSCODE_pd.Series.to_frame(df).iloc[start:end]
     elif isinstance(df, dict):
         df = _VSCODE_pd.Series(df)
-        df = _VSCODE_pd.Series.to_frame(df)
+        df = _VSCODE_pd.Series.to_frame(df).iloc[start:end]
     elif hasattr(df, "toPandas"):
-        df = df.toPandas()
+        df = df.toPandas().iloc[start:end]
```
Comment (joyceerhl) [inline pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:101]: Broke chunking in https://github.com/microsoft/vscode-jupyter/pull/4951/

File: src/client/datascience/jupyter/kernelVariables.ts [hunk 2]
```diff
@@ -164,10 +164,6 @@ export class KernelVariables implements IJupyterVariables {
         // Import the data frame script directory if we haven't already
         await this.importDataFrameScripts(notebook);
 
-        if (targetVariable.rowCount) {
-            end = Math.min(end, targetVariable.rowCount);
-        }
```
Comment (joyceerhl) [inline src/client/datascience/jupyter/kernelVariables.ts:169]: The row start and end however are already evaluated relative to the slice variable info, so this check is unnecessary.

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py (hunk 1 above)
Comment (DonJayamanne) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:101]: Sorry, but the comment doesn't make sense.
It looks like this is a bug fix.

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py [hunk 3]
```diff
@@ -91,14 +91,14 @@ def _VSCODE_convertTensorToDataFrame(tensor, start=None, end=None):
 def _VSCODE_convertToDataFrame(df, start=None, end=None):
     vartype = type(df)
     if isinstance(df, list):
-        df = _VSCODE_pd.DataFrame(df)
+        df = _VSCODE_pd.DataFrame(df).iloc[start:end]
```
Comment (rchiodo) [inline pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:94]: Do these work if start and end are none?

**Review by rchiodo — APPROVED** [review]

File: src/datascience-ui/plot/mainPanel.tsx [hunk 4]
```diff
@@ -21,6 +21,9 @@ import { SvgViewer } from '../react-common/svgViewer';
 import { TestSvg } from './testSvg';
 import { Toolbar } from './toolbar';
 
+import { initializeIcons } from '@fluentui/react';
```
Comment (IanMatthewHuff) [inline src/datascience-ui/plot/mainPanel.tsx:24]: Comment might be good here (I know this is just a move). Not really obvious to me what this is doing.

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py (hunk 1 above)
Comment (IanMatthewHuff) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:101]: I'm a tiny bit confused as well. This is merging into more-styles and currently there is also a PR open for more-styles into main?

Comment (joyceerhl) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:101]: Yeah this PR depends on the Checkbox change in more-styles. But I'm merging the style changes into main, then this into main. And yes this is a bug fix.

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py (hunk 3 above)
Comment (joyceerhl) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:94]: Yup, that corresponds to no slice i.e. it returns the full variable

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py (hunk 1 above)
Comment (IanMatthewHuff) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:101]: Ok, that works for me. I think that personally I would just park it until style was into main and then PR against main so I could see test runs and what not. But small enough change that I'm down with approving now.

**Review by IanMatthewHuff — APPROVED** [review]

File: pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py (hunk 1 above)
Comment (joyceerhl) [reply pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py:101]: Somehow I thought we used to get test runs even on PRs not going into main... That's totally fair, my bad. Can't wait for GitHub to add stacked diffs 😭

**Pull request merged**